crates/api_models/src/payments.rs

```diff
@@ -6457,6 +6459,23 @@ pub struct SdkInformation {
     pub sdk_max_timeout: u8,
 }
 
+#[cfg(feature = "v2")]
+#[derive(Debug, serde::Serialize, serde::Deserialize, Clone, ToSchema)]
+pub struct PaymentMethodsListRequest {}
+
+#[cfg(feature = "v2")]
+#[derive(Debug, serde::Serialize, ToSchema)]
+pub struct PaymentMethodListResponseForPayments {
+    /// The list of payment methods that are enabled for the business profile
+    #[schema(value_type = Vec<ResponsePaymentMethodTypes>)]
+    pub payment_methods_enabled: Vec<payment_methods::ResponsePaymentMethodTypes>,
+
+    /// The list of payment methods that are saved by the given customer
+    /// This field is only returned if the customer_id is provided in the request
+    #[schema(value_type = Vec<CustomerPaymentMethod>)]
+    pub customer_payment_methods: Vec<payment_methods::CustomerPaymentMethod>,
+}
```

Comment:
In case the customer_id is not provided, an empty vector should be returned then? 
Should we consider using Option here?

Reply:
Yes, the inference would be based on the option then. Have made this into an option. But I think we should be explicit about this. Will change this based on the discussions if required later